app/controllers/registrations_controller.rb

```diff
@@ -34,6 +34,53 @@ class RegistrationsController < ApplicationController
     redirect_to competition_path(competition_from_params) if competition_from_params.use_wca_registration?
   end
 
+  before_action :check_errors_import_registration!, only: %i[do_import]
+  private def check_errors_import_registration!
+    @competition = competition_from_params
+    file = params[:csv_registration_file]
+
+    competition_events = @competition.competition_events
+    required_event_columns = competition_events.map(&:event_id)
```

Comment:
```suggestion
    required_event_columns = competition_events.pluck(:event_id)
```

If you're working on an ActiveRecord relation, `pluck` is always your friend for some individual field